src/disco/components/Addon.js

```diff
@@ -279,7 +288,13 @@ export function mapDispatchToProps(dispatch, { _tracking = tracking,
           () => {
             log.info(`Add-on "${guid}" not found so setting status to UNINSTALLED`);
             dispatch({ type: INSTALL_STATE, payload: { ...payload, status: UNINSTALLED } });
-          });
+          }
+        )
+        .catch((err) => {
+          log.error(err);
+          // Dispatch a generic error should the success/error functions throw.
+          dispatch({ type: INSTALL_STATE, payload: { guid, status: ERROR, error: FATAL_ERROR } });
```

Comment:
Style nitpick here, but this is on one line and the others are multi-line (which I find easier to read).

Might just be nice to be consistent, but I'll let you pick which.